pkg/services/accesscontrol/resourcepermissions/store.go

```diff
@@ -775,6 +803,9 @@ func (s *InMemoryActionSets) ExpandActionSets(permissions []accesscontrol.Permis
 			continue
 		}
 		for _, action := range resolvedActions {
+			if actionFilter != "" && !strings.HasPrefix(action, actionFilter) {
```

Comment:
I don't think we have the case anywhere, but this might match more than one action when action filter is an action (unlikely ex: `datasources:read` `datasources:read-id`, when searching `datasources:read`).

Reply:
You're right. I'll extend this to do an exact match for actions. Feels a bit meh to add additional complexity for a potential case, but I think we might actually have it already with some alert actions, so better safe than sorry.